File: Mathematics/palindrome/java/palindrome.java
import java.util.*;
import java.io.*;

public class palindrome {
	public static void main(String args[]) {
		int num,d,r,n;
		Scanner s = new Scanner(System.in);
		r=0;
		System.out.println("Enter a number: ");
		num = s.nextInt();
		n = num;
		do {
			d = num%10;
			r = (r*10) + d;
			num = num/10;
		} while(num!=0);

		System.out.println(n+"  "+r);

		if(n==r) 
			System.out.println("It is a palindrome");
		else
			System.out.println("It is not a palindrome");

	}
}

File: Mathematics/palindrome/c/palindrome.c
#include <stdio.h>

int reverse(int x){
	int reverse = 0;
	while (x){
		reverse = reverse * 10 + (x % 10);
		x /= 10;
	}
	return reverse;
}

void palindrome(int x){
	if (x == reverse(x)) 
		printf("This number is a palindrome !\n");
	else
		printf("This number is not a palindrome !\n");
}

int main() {
	int x;
	scanf("%d", &x);
	palindrome(x); 
	return 0;
}

